src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -109,6 +110,20 @@ public void channelConnected(ChannelHandlerContext ctx,
                     zkServer, NettyServerCnxnFactory.this);
             ctx.setAttachment(cnxn);
 
+            InetAddress addr = ((InetSocketAddress)cnxn.channel.getRemoteAddress()).getAddress();
+            Set<NettyServerCnxn> s = ipMap.get(addr);
+            if (s != null) {
+                synchronized (s) {
+                    int cnxncount = s.size();
+                    if (maxClientCnxns > 0 && cnxncount >= maxClientCnxns) {
```

Comment:
We disable `maxClientCnxns` by setting it to zero. The documentation says:
_Setting this to 0 entirely removes the limit on concurrent connections._